src/browser/setupWorker/start/createRequestListener.ts

```diff
@@ -47,18 +48,29 @@ export const createRequestListener = (
             // ".log()" method of the request handler.
             const responseClone = response.clone()
             const responseInit = toResponseInit(response)
-            const responseStream = responseClone.body
 
-            messageChannel.postMessage(
-              'MOCK_RESPONSE',
-              {
+            /**
+             * @note Safari doesn't support transferring a "ReadableStream".
+             * Check that the browser supports that before sending it to the worker.
+             */
+            if (supportsReadableStreamTransfer()) {
+              const responseStream = response.body
+              messageChannel.postMessage(
+                'MOCK_RESPONSE',
+                {
+                  ...responseInit,
+                  body: responseStream,
+                },
+                responseStream ? [responseStream] : undefined,
+              )
+            } else {
+              // As a fallback, send the response body buffer to the worker.
+              const responseBuffer = await responseClone.arrayBuffer()
```

Comment:
@thepassle, would this work for your use case if we read the mocked response body as `ArrayBuffer` instead of `Blob`? I find it to be more correct.

Reply:
looks like it works: 
<img width="1020" alt="image" src="https://github.com/mswjs/msw/assets/17054057/6d6722b5-c0ce-4253-8d94-d7bfa3e6be7f">

Reply:
Awesome! 

In the future, we should print a warning if the developer is mocking a `ReadableStream` body while transferring that stream isn't supported by their browser. We will still use the fallback to `ArrayBuffer` but the developer may be confused lest we show a warning of what's happening.